middleware/reportflow.go

```diff
@@ -0,0 +1,35 @@
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/goyek/goyek/v2"
+)
+
+// ReportStatus is a middleware which reports the flow execution status.
+//
+// The format is based on the reports provided by the Go test runner.
+func ReportFlow(next goyek.Executor) goyek.Executor {
+	return func(in goyek.ExecuteInput) error {
+		out := in.Output
+
+		from := time.Now()
+		err := next(in)
+		if _, ok := err.(*goyek.FailError); ok {
+			fmt.Fprintf(out, "%v\t%.3fs\n", err, time.Since(from).Seconds())
```

Comment:
Usually errors are reported to stderr

Same remark for other error reporting made in this file

Reply:
This uses the output configured by `Flow.SetOutput`. The same output is used e.g. by `middleware.ReportStatus`, `middleware.ReportLongRun` and this output is also passed to `Logger`.

There was an idea to have two output streams (output and error), but it was rejected when working on `v2` for sake of simplification.

Reply:
OK.